crates/matrix-sdk/src/room/joined/futures.rs

```diff
@@ -0,0 +1,218 @@
+#[cfg(feature = "image-proc")]
+use std::io::Cursor;
+use std::{
+    future::{Future, IntoFuture},
+    pin::Pin,
+};
+
+use eyeball::shared::Observable as SharedObservable;
+use mime::Mime;
+use ruma::api::client::message::send_message_event;
+use tracing::{Instrument, Span};
+
+use super::Joined;
+use crate::{attachment::AttachmentConfig, Result, TransmissionProgress};
+#[cfg(feature = "image-proc")]
+use crate::{
+    attachment::{generate_image_thumbnail, Thumbnail},
+    error::ImageError,
+};
+
+/* pub struct PrepareAndSendAttachment<'a> {
```

Comment:
What's this about?

Reply:
Ah, this is outdated. Since `prepare_and_send_attachment` is an internal function I ended up adding an extra `send_progress` parameter which is simpler than the builder-style API that `send_attachment()` got.